src/DescriptorsCommand.cc

```diff
@@ -735,16 +734,51 @@ Json::Value FindDescriptor::construct_responses(
 
         findDesc = json_responses[0];
 
-        if (findDesc.isMember("entities")) {
-            convert_properties(findDesc["entities"], list);
-        }
-
         if (findDesc["status"] != 0) {
             Json::Value return_error;
             return_error["status"]  = RSCommand::Error;
             return_error["info"]    = "Descriptors Not Found";
             return error(return_error);
         }
+
+        const Json::Value& set_response = json_responses[1];
+        const Json::Value& set = set_response["entities"][0];
+
+        // These properties should always exist
+        assert(set.isMember(VDMS_DESC_SET_PATH_PROP));
+        assert(set.isMember(VDMS_DESC_SET_DIM_PROP));
+        std::string set_path = set[VDMS_DESC_SET_PATH_PROP].asString();
+        int dim = set[VDMS_DESC_SET_DIM_PROP].asInt();
+
+        if (findDesc.isMember("entities")) {
+
+            if (get_value<bool>(results, "blob", false)) {
```

Comment:
Isn't this if {} block triplicated in the three conditions? Good time to consolidate?